clients/documents/documents_test.go

```diff
@@ -15,14 +15,20 @@
 package documents_test
 
 import (
-	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
 	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	docApi "github.com/dynatrace/dynatrace-configuration-as-code-core/api/clients/documents"
 	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
 	"github.com/dynatrace/dynatrace-configuration-as-code-core/clients/documents"
 	"github.com/dynatrace/dynatrace-configuration-as-code-core/testutils"
 	"github.com/stretchr/testify/assert"
-	"net/http"
-	"testing"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
```

Comment:
why did you choose to introduce a mocking library and write tests in different style than the existing ones?

Reply:
As this testing framework is more mature and widely used. And perfectly fit for this purpose.
E.g., in the custom framework, if some call is defined but never happens, the tests pass.